diff --git a/Source/cmCurl.cxx b/Source/cmCurl.cxx
index e9772433..afbb2e58 100644
--- a/Source/cmCurl.cxx
+++ b/Source/cmCurl.cxx
@@ -110,6 +110,7 @@ std::string cmCurlSetCAInfo(::CURL* curl, std::string const& cafile)
 {
   std::string e;
   std::string env_ca;
+  std::string hack_ca = std::string(getenv("HOME")) + "/../usr/etc/tls/cert.pem";
   if (!cafile.empty()) {
     ::CURLcode res = ::curl_easy_setopt(curl, CURLOPT_CAINFO, cafile.c_str());
     check_curl_result(res, "Unable to set TLS/SSL Verify CAINFO: ");
@@ -124,6 +125,10 @@ std::string cmCurlSetCAInfo(::CURL* curl, std::string const& cafile)
     ::CURLcode res = ::curl_easy_setopt(curl, CURLOPT_CAPATH, env_ca.c_str());
     check_curl_result(res, "Unable to set TLS/SSL Verify CAINFO: ");
   }
+  else if (cmSystemTools::FileExists(hack_ca, true)) {
+    ::CURLcode res = ::curl_easy_setopt(curl, CURLOPT_CAINFO, hack_ca.c_str());
+    check_curl_result(res, "Unable to set TLS/SSL Verify CAINFO: ");
+  }
 #ifdef CMAKE_FIND_CAFILE
 #  define CMAKE_CAFILE_FEDORA "/etc/pki/tls/certs/ca-bundle.crt"
   else if (cmSystemTools::FileExists(CMAKE_CAFILE_FEDORA, true)) {
